jazkarta/shop/browser/templates/shipping_methods_controlpanel.pt

```diff
@@ -12,7 +12,14 @@
   <div data-pat-plone-modal='{"actionOptions": {"displayInModal": false}}'>
     <h1 class="documentFirstHeading">Shipping Methods</h1>
 
-    <a class="pat-plone-modal green button" href="${context/absolute_url}/${view/__name__}/+"><button>Add Shipping Method</button></a>
+    <tal:using_p5 tal:condition="view/using_plone5">
+        <a class="pat-plone-modal green button" href="${context/absolute_url}/${view/__name__}/+"><button>Add Shipping Method</button></a>
+    </tal:using_p5>
+    <tal:using_p4 tal:condition="not:view/using_plone5">
+        <a class="p4-modal" 
+                  tal:attributes="href string:${context/absolute_url}/${view/__name__}/+"><button>Add Shipping Method</button></a>
+    </tal:using_p4>
```

Comment:
Why not always use the pat-plone-modal class, and make p4-overlays.js look for that rather than p4-modal?